contracts/contracts/strategies/qf-maci/MACIQF.sol

```diff
@@ -227,15 +221,17 @@ contract MACIQF is MACIQFBase, DomainObjs, Params {
             bytes memory _proof
         ) = abi.decode(_data, (PubKey, uint256, bytes));
 
-        if (isAddressZero(_maci)) revert MaciNotSet();
+        if (_isAddressZero(maci)) revert MaciNotSet();
         if (isFinalized) revert RoundAlreadyFinalized();
         if (contributorInfo[_sender].signedUp) revert AlreadyContributed();
         if (amount > MAX_VOICE_CREDITS * voiceCreditFactor) revert ContributionAmountTooLarge();
 
         // Validate allowlist proof if provided GAS optimization
         // Don't check if the proof is empty
         if (_proof.length != 0) {
-            if (!allowlistVerifier.validateAllowlist(_proof)) {
+
+            uint256 watermark = allowlistVerifier.validateAllowlist(_proof);
+            if (watermark != 0 && watermark != uint256(uint160(_sender))) {
```

Comment:
If `watermark == 0` then we won't revert here, but in `validateAllowlist`, we return 0 if the validation fails. Thus it seems like this allows invalid proofs to succeed. I think instead what we want here is:

```solidity
if (watermark == 0 || watermark != uint256(uint160(_sender))) {
  revert InvalidProof();
}
```

Which we can simplify, since the `_sender` can't possibly be `address(0)`, as:

```solidity
if (watermark != uint256(uint160(_sender))) {
  revert InvalidProof();
}
```